client/txn.go

```diff
@@ -407,8 +419,20 @@ func (txn *Txn) CommitBy(deadline roachpb.Timestamp) *roachpb.Error {
 }
 
 // Rollback sends an EndTransactionRequest with Commit=false.
+// The txn's status is set to ABORTED in case of error. txn is
+// considered finalized and cannot be used to send any more commands.
 func (txn *Txn) Rollback() *roachpb.Error {
-	return txn.sendEndTxnReq(false /* commit */, nil)
+	err := txn.sendEndTxnReq(false /* commit */, nil)
+	if err != nil {
+		// TODO(andrei): I'm not completely sure setting the status here is required;
```

Comment:
`txn.Proto.Status = ABORTED` is not set on the `send()` return path unless the error experienced was `TransactionAbortedError` (`txn.Proto` is only updated on abort and retry errors). It seems reasonable to set this here. Remove the TODO and add a comment mentioning the desire to poison the txn on a failure to rollback.